providers/implementations/kem/mlx_kem.c

```diff
@@ -167,7 +190,9 @@ static int mlx_kem_encapsulate(void *vctx, unsigned char *ctext, size_t *clen,
     encap_slen = ML_KEM_SHARED_SECRET_BYTES;
     cbuf = ctext + ml_kem_slot * key->xinfo->pubkey_bytes;
     sbuf = shsec + ml_kem_slot * key->xinfo->shsec_bytes;
-    ctx = EVP_PKEY_CTX_new_from_pkey(key->libctx, key->mkey, key->propq);
+    adjusted_propq = get_adjusted_propq(key->propq);
+    ctx = EVP_PKEY_CTX_new_from_pkey(key->libctx, key->mkey,
```

Comment:
The only context in which adjusted_propq is NULL is when malloc fails - and that should cause this function to return an error - not be silently ignored - as that outcome is unlikely to be what the user wants - check null and return an error.